File: pyg_parent/pyg_cart_web_91/src/main/webapp/js/service/cartService.js
//服务层
app.service('cartService',function($http){
	    	
	//将商品加入购物车方法
	this.addItemToCartList=function(itemId, num){
		return $http.get('../cart/addItemToCartList/'+itemId+"/"+num);
	}
	//查询购物车列表
    this.findCartList = function () {
        return $http.get('../cart/findCartList');
    }
});

File: pyg_parent/pyg_cart_91/pyg_cart_interface_91/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.vo.Cart;

import java.util.List;

public interface CartService {
    List<Cart> findCartListByKey(String key);

    List<Cart> addItemToCartList(List<Cart> cartList, Long itemId, Integer num);

    void saveCartListToRedis(List<Cart> cartList, String key);

    List<Cart> mergeList(List<Cart> userCartList, List<Cart> sessionCartList);

    void deleteCartListByKey(String key);
}
